worlds/checksmate/__init__.py

```diff
@@ -0,0 +1,386 @@
+        # Remarkable Rookies
+        2: {"Progressive Minor Piece": 2, "Progressive Major Piece": 1, "Progressive Major To Queen": 1},
+        # Nutty Knights (although the Short Rook and Half Duck swap potency)
+        3: {"Progressive Minor Piece": 2, "Progressive Major Piece": 1, "Progressive Major To Queen": 1},
+        # Eurasian pieces
+        4: {"Progressive Minor Piece": 2, "Progressive Major Piece": 1, "Progressive Major To Queen": 1},
+    }
+
+    def __init__(self, multiworld: MultiWorld, player: int):
+        super(CMWorld, self).__init__(multiworld, player)
+        self.locked_locations = []
+
+    def generate_early(self) -> None:
+        army_constraint = get_option_value(self.multiworld, self.player, "fairy_chess_army")
+        if army_constraint != 0:
+            which_pieces = get_option_value(self.multiworld, self.player, "fairy_chess_pieces")
+            # Full: Adds the 12 Chess With Different Armies pieces, the Cannon, and the Vao.
+            if which_pieces == 1:
+                army_options = [0, 1, 2, 3, 4]
+            # CwDA: Adds the pieces from Ralph Betza's 12 Chess With Different Armies.
+            elif which_pieces == 2:
+                army_options = [0, 1, 2, 3]
+            # Cannon: Adds a Rook-like piece, which captures a distal chessman by leaping over an intervening chessman.
+            # Eurasian: Adds the Cannon and the Vao, a Bishop-like Cannon, in that it moves and captures diagonally.
+            elif which_pieces > 2:
+                army_options = [0, 4]
+            # Vanilla: Disables fairy chess pieces completely.
+            else:
+                army_options = [0]
+            self.army[self.player] = self.multiworld.per_slot_randoms[self.player].choice(army_options)
+        is_single = get_option_value(self.multiworld, self.player, "goal") == 0
+        if not is_single:
+            return
+        for enemy_pawn in self.item_name_groups["Enemy Pawn"]:
+            self.multiworld.start_inventory[self.player].value[enemy_pawn] = 1
+        for enemy_piece in self.item_name_groups["Enemy Piece"]:
+            self.multiworld.start_inventory[self.player].value[enemy_piece] = 1
+
+    def setting(self, name: str):
+        return getattr(self.multiworld, name)[self.player]
+
+    def fill_slot_data(self) -> dict:
+        cursed_knowledge = {name: self.multiworld.per_slot_randoms[self.player].getrandbits(31) for name in [
+            "pocket_seed", "pawn_seed", "minor_seed", "major_seed", "queen_seed"]}
+        potential_pockets = [0,0,0,0,1,1,1,1,2,2,2,2]
+        self.multiworld.per_slot_randoms[self.player].shuffle(potential_pockets)
+        cursed_knowledge["pocket_order"] = potential_pockets
+        if self.player in self.army:
+            cursed_knowledge["army"] = self.army[self.player]
+        return dict(cursed_knowledge, **{option_name: self.setting(option_name).value for option_name in cm_options})
```

Comment:
You probably do, but make sure you are only sending the information that the client/any potential trackers need.

Reply:
also `self.options.as_dict(option_names)`

Reply:
Thanks treble.

Silvris, the seeds are used to generate specific pieces and locations in the client when either option is set to stable. However, this could be omitted if both are set to chaos. Let me know if this is what you're referring to.

Reply:
I believe I have reconciled this issue.